### PR TITLE
Switch plugin arg order & add context

### PR DESCRIPTION
**Pull request by neftaly**

### CODE REVIEW

File: src/plugins.js [hunk 1]
```diff
@@ -3,10 +3,10 @@
 export const inMemory = (initial : Object, transition : Function) => {
   let rootState = initial;
   const read = () => rootState;
-  const write = (fn : Function) => {
+  const write = (fn : Function, ...context: any) => {
```
Comment (neftaly) [inline src/plugins.js:6]: I don't know if this is correct for rest operators in flow

**Comment by neftaly** [comment]
If you're happy with this, can you please bump the version and publish? As it's 0.0.x, it'll be semver-compatible to go to 0.0.3 with a breaking change. 0.1.0 might be async watchers + new tests.

File: src/plugins.js (hunk 1 above)
Comment (jameshopkins) [reply src/plugins.js:6]: I'm actually not sure myself :) Will have a look tomorrow

**Comment by jameshopkins** [comment]
Can you update the [relevant documentation](https://github.com/jameshopkins/atom-store#watchfnprevvalue-nextvalue-) with the arg order changes and a small description about adding arbitrary `context`-related arguments?

**Comment by neftaly** [comment]
Sorry, I forgot to do that

**Comment by jameshopkins** [comment]
Broadly, this looks good. I'll take a closer look at it tomorrow and prepare a release

**Comment by neftaly** [comment]
Cheers!

**Pull request merged**